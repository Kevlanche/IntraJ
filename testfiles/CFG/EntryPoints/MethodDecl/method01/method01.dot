digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[6]"; "Opt[9]"; "List[10]"; "List[11]"}
{ rank = same; "List[7]"; "MethodDecl[12]"; "MethodDecl[38]"; "InstanceInitializer[53]"}
{ rank = same; "Modifier[8]"; "Entry[13]"; "Modifiers[14]"; "PrimitiveTypeAccess[17]"; "List[18]"; "List[19]"; "Opt[20]"; "Exit[21]"; "Entry[39]"; "Modifiers[40]"; "TypeAccess[43]"; "List[44]"; "List[45]"; "Opt[46]"; "Exit[47]"; "Block[54]"}
{ rank = same; "List[15]"; "Block[21]"; "List[41]"; "Block[47]"; "List[55]"}
{ rank = same; "Modifier[16]"; "List[22]"; "Modifier[42]"; "List[48]"}
{ rank = same; "ExprStmt[23]"; "ReturnStmt[49]"}
{ rank = same; "Dot[24]"; "Opt[50]"; "List[52]"; "UncheckedExceptions[50]"}
{ rank = same; "Dot[25]"; "MethodAccess[28]"; "BooleanLiteral[51]"}
{ rank = same; "TypeAccess[26]"; "VarAccess[27]"; "List[30]"; "List[36]"; "List[37]"; "UncheckedExceptions[29]"}
{ rank = same; "MethodAccess[31]"}
{ rank = same; "List[33]"; "List[34]"; "List[35]"; "UncheckedExceptions[32]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < method01 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[8]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[9]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[10]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[12]" [label="MethodDecl\n < test >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[13]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[14]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[15]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[16]" [label="Modifier\n < private >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[17]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[18]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[19]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[20]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[21]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[22]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[23]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[24]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[25]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[26]" [label="TypeAccess\n < System >", fillcolor=white   style=filled  ]
"VarAccess[27]" [label="VarAccess\n < out >", fillcolor=white   style=filled  ]
"MethodAccess[28]" [label="MethodAccess\n < println >", fillcolor=white   style=filled  ]
"List[30]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodAccess[31]" [label="MethodAccess\n < F >", fillcolor=white   style=filled  ]
"List[33]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[34]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[35]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[32]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[36]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[37]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[29]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[21]" [label="Exit\n ", fillcolor=white   style=filled  ]
"MethodDecl[38]" [label="MethodDecl\n < F >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[39]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[40]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[41]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[42]" [label="Modifier\n < private >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[43]" [label="TypeAccess\n < Boolean >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[44]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[45]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[46]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[47]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[48]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ReturnStmt[49]" [label="ReturnStmt\n ", fillcolor=white   style=filled  ]
"Opt[50]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"BooleanLiteral[51]" [label="BooleanLiteral\n < false >", fillcolor=white   style=filled  ]
"List[52]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[50]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[47]" [label="Exit\n ", fillcolor=white   style=filled  ]
"InstanceInitializer[53]" [label="InstanceInitializer\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[54]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[55]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[7]" -> "Modifier[8]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[9]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[10]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Entry[13]" [style=dashed, color=gray]
"List[15]" -> "Modifier[16]" [style=dashed, color=gray]
"Modifiers[14]" -> "List[15]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Modifiers[14]" [style=dashed, color=gray]
"MethodDecl[12]" -> "PrimitiveTypeAccess[17]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[18]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[19]" [style=dashed, color=gray]
"Dot[25]" -> "TypeAccess[26]" [style=dashed, color=gray]
"Dot[25]" -> "VarAccess[27]" [style=dashed, color=gray]
"Dot[24]" -> "Dot[25]" [style=dashed, color=gray]
"MethodAccess[31]" -> "List[33]" [style=dashed, color=gray]
"MethodAccess[31]" -> "List[34]" [style=dashed, color=gray]
"MethodAccess[31]" -> "List[35]" [style=dashed, color=gray]
"MethodAccess[31]" -> "UncheckedExceptions[32]" [style=dashed, color=gray]
"List[30]" -> "MethodAccess[31]" [style=dashed, color=gray]
"MethodAccess[28]" -> "List[30]" [style=dashed, color=gray]
"MethodAccess[28]" -> "List[36]" [style=dashed, color=gray]
"MethodAccess[28]" -> "List[37]" [style=dashed, color=gray]
"MethodAccess[28]" -> "UncheckedExceptions[29]" [style=dashed, color=gray]
"Dot[24]" -> "MethodAccess[28]" [style=dashed, color=gray]
"ExprStmt[23]" -> "Dot[24]" [style=dashed, color=gray]
"List[22]" -> "ExprStmt[23]" [style=dashed, color=gray]
"Block[21]" -> "List[22]" [style=dashed, color=gray]
"Opt[20]" -> "Block[21]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Opt[20]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Exit[21]" [style=dashed, color=gray]
"List[11]" -> "MethodDecl[12]" [style=dashed, color=gray]
"MethodDecl[38]" -> "Entry[39]" [style=dashed, color=gray]
"List[41]" -> "Modifier[42]" [style=dashed, color=gray]
"Modifiers[40]" -> "List[41]" [style=dashed, color=gray]
"MethodDecl[38]" -> "Modifiers[40]" [style=dashed, color=gray]
"MethodDecl[38]" -> "TypeAccess[43]" [style=dashed, color=gray]
"MethodDecl[38]" -> "List[44]" [style=dashed, color=gray]
"MethodDecl[38]" -> "List[45]" [style=dashed, color=gray]
"Opt[50]" -> "BooleanLiteral[51]" [style=dashed, color=gray]
"ReturnStmt[49]" -> "Opt[50]" [style=dashed, color=gray]
"ReturnStmt[49]" -> "List[52]" [style=dashed, color=gray]
"ReturnStmt[49]" -> "UncheckedExceptions[50]" [style=dashed, color=gray]
"List[48]" -> "ReturnStmt[49]" [style=dashed, color=gray]
"Block[47]" -> "List[48]" [style=dashed, color=gray]
"Opt[46]" -> "Block[47]" [style=dashed, color=gray]
"MethodDecl[38]" -> "Opt[46]" [style=dashed, color=gray]
"MethodDecl[38]" -> "Exit[47]" [style=dashed, color=gray]
"List[11]" -> "MethodDecl[38]" [style=dashed, color=gray]
"Block[54]" -> "List[55]" [style=dashed, color=gray]
"InstanceInitializer[53]" -> "Block[54]" [style=dashed, color=gray]
"List[11]" -> "InstanceInitializer[53]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[11]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[21]" -> "MethodAccess[28]" [color=red, constraint=false]
"MethodAccess[28]" -> "Exit[21]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[28]" -> "MethodAccess[31]" [color=red, constraint=false]
"MethodAccess[31]" -> "MethodAccess[28]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[31]" -> "VarAccess[27]" [color=red, constraint=false]
"VarAccess[27]" -> "MethodAccess[31]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[27]" -> "TypeAccess[26]" [color=red, constraint=false]
"TypeAccess[26]" -> "VarAccess[27]" [color=blue, constraint=false, xlabel=" "] 
"TypeAccess[26]" -> "Entry[13]" [color=red, constraint=false]
"Entry[13]" -> "TypeAccess[26]" [color=blue, constraint=false, xlabel=" "] 
"Exit[47]" -> "ReturnStmt[49]" [color=red, constraint=false]
"ReturnStmt[49]" -> "Exit[47]" [color=blue, constraint=false, xlabel=" "] 
"ReturnStmt[49]" -> "BooleanLiteral[51]" [color=red, constraint=false]
"BooleanLiteral[51]" -> "ReturnStmt[49]" [color=blue, constraint=false, xlabel=" "] 
"BooleanLiteral[51]" -> "Entry[39]" [color=red, constraint=false]
"Entry[39]" -> "BooleanLiteral[51]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"Modifiers[6]" -> "Opt[9]" -> "List[10]" -> "List[11]" [style=invis]
"MethodDecl[12]" -> "MethodDecl[38]" -> "InstanceInitializer[53]" [style=invis]
"Entry[13]" -> "Modifiers[14]" -> "PrimitiveTypeAccess[17]" -> "List[18]" -> "List[19]" -> "Opt[20]" -> "Exit[21]" [style=invis]
"Dot[25]" -> "MethodAccess[28]" [style=invis]
"TypeAccess[26]" -> "VarAccess[27]" [style=invis]
"Entry[39]" -> "Modifiers[40]" -> "TypeAccess[43]" -> "List[44]" -> "List[45]" -> "Opt[46]" -> "Exit[47]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
59", fillcolor=1]
Edges [label="Total Nr. Edges
74", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
9", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
8", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}