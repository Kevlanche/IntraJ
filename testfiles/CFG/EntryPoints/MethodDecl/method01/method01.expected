NodeName: Entry[13] Successors: [TypeAccess[26]]
NodeName: TypeAccess[26] Successors: [VarAccess[27]]
NodeName: VarAccess[27] Successors: [MethodAccess[31]]
NodeName: MethodAccess[31] Successors: [MethodAccess[28]]
NodeName: MethodAccess[28] Successors: [Exit[21]]
NodeName: Exit[21] Successors: []
NodeName: Entry[39] Successors: [BooleanLiteral[51]]
NodeName: BooleanLiteral[51] Successors: [ReturnStmt[49]]
NodeName: ReturnStmt[49] Successors: [Exit[47]]
NodeName: Exit[47] Successors: []

NodeName: Exit[21] Predecessor: [MethodAccess[28]]
NodeName: MethodAccess[28] Predecessor: [MethodAccess[31]]
NodeName: MethodAccess[31] Predecessor: [VarAccess[27]]
NodeName: VarAccess[27] Predecessor: [TypeAccess[26]]
NodeName: TypeAccess[26] Predecessor: [Entry[13]]
NodeName: Entry[13] Predecessor: []
NodeName: Exit[47] Predecessor: [ReturnStmt[49]]
NodeName: ReturnStmt[49] Predecessor: [BooleanLiteral[51]]
NodeName: BooleanLiteral[51] Predecessor: [Entry[39]]
NodeName: Entry[39] Predecessor: []
